tests/test_otel.py

```diff
@@ -0,0 +1,524 @@
+import binascii
+from conftest import self_signed_cert
+import niquests
+import pytest
+import socket
+import ssl
+import subprocess
+import time
+import urllib3
+
+
+CERTS = [self_signed_cert, "localhost"]
+
+NGINX_CONFIG = """
+{{ globals }}
+
+daemon off;
+
+events {
+}
+
+http {
+    {{ http_globals }}
+
+    ssl_certificate localhost.crt;
+    ssl_certificate_key localhost.key;
+
+    otel_exporter {
+        endpoint 127.0.0.1:14317;
+        interval 1s;
+        batch_size 10;
+        batch_count 2;
+    }
+
+    otel_trace on;
+    otel_service_name {{ name }};
+
+    add_header "X-Otel-Trace-Id" $otel_trace_id;
+    add_header "X-Otel-Span-Id" $otel_span_id;
+    add_header "X-Otel-Parent-Id" $otel_parent_id;
+    add_header "X-Otel-Parent-Sampled" $otel_parent_sampled;
+
+    server {
+        listen       127.0.0.1:18443 {{ mode }};
```

Comment:
"http2" flag of listen is deprecated, we should use separate directive instead.

Reply:
Fixed.